src/test/java/org/opensearch/security/securityconf/SecurityRolesPermissionsTest.java

```diff
@@ -221,6 +223,14 @@ public void hasExplicitClusterPermissionPermissionForRestAdmin() {
         assertHasNoPermissionsForRestApiAdminOnePermissionRole(Endpoint.CONFIG, securityConfigAllowRole);
     }
 
+    @Test
+    public void testCreateSecurityRole() {
+        InMemorySecurityRoles securityRoles = new InMemorySecurityRolesV7(1);
+        securityRoles.addSecurityRole("testRole", Set.of("cluster:monitor/health"), Map.of("*", Set.of("indices:data/read/search")));
+        assertTrue(securityRoles.getRoleNames().contains("testRole"));
+        assertTrue(securityRoles.hasExplicitClusterPermissionPermission("cluster:monitor/health"));
```

Comment:
Minor nit: we should also check that index permissions are working as expected

Reply:
Index permissions aren't supported in this PR. I opened up a [PR in core](https://github.com/opensearch-project/OpenSearch/pull/15778) to create a mechanism for plugins to request index permissions. This PR will support a majority of existing plugin use-cases of direct access to their own system indices.

For plugins, like the security plugin, that need to perform actions on regular indices they still need to use `ThreadContext.stashContext()` until index permissions are supported.